lib/archethic/mining/pending_transaction_validation.ex

```diff
@@ -159,6 +159,28 @@ defmodule Archethic.Mining.PendingTransactionValidation do
     end
   end
 
+  defp do_accept_transaction(%Transaction{
+         type: :origin,
+         data: %TransactionData{
+           content: content
+         }
+       }) do
+    with {<<_curve_id::8, _origin_id::8, origin_public_key::binary>>, key_certificate} <-
```

Comment:
I think you can extract the certificate like this: https://github.com/archethic-foundation/archethic-node/blob/b8a0952e1a5e624e5c9f0ead0da898bdf8bbd195/lib/archethic/p2p/node.ex#L87 
by encoding the certificate size we can have a better control of the deserialized data